src/client/theme-default/components/VPDocAsideOutline.vue

```diff
@@ -1,21 +1,56 @@
 <script setup lang="ts">
-import { ref } from 'vue'
-import { useData } from 'vitepress'
+import type { DefaultTheme } from 'vitepress/theme'
+import { useData, useRoute } from 'vitepress'
+import { computed, ref, watch } from 'vue'
+import VPDocAsideOutlineItem from './VPDocAsideOutlineItem.vue'
 import {
-  useOutline,
-  useActiveAnchor
+  useActiveAnchor,
+  resolveHeaders,
+  type MenuItem
 } from '../composables/outline.js'
 
-const { page, frontmatter, theme } = useData()
-
-const { hasOutline } = useOutline()
+const { frontmatter, theme } = useData()
+const route = useRoute()
+
+const pageOutline = computed<DefaultTheme.Config['outline']>(
+  () => frontmatter.value.outline ?? theme.value.outline
+)
+
+const headers = ref<MenuItem[]>([])
+
+watch(
```

Comment:
This watcher thing is currently running on route change. It won't work if some content is manipulated later or a node is added/removed. Need to use MutationObserver here, but things are getting complex in that.